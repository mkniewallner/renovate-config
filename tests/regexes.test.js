import { extractPackageFile } from "renovate/dist/modules/manager/regex";
import { loadFixture, loadRenovateConfiguration } from "./utils";

it("should find YAML dependencies", () => {
  const fileName = "ci.yml";

  const res = extractPackageFile(
    loadFixture(fileName),
    fileName,
    loadRenovateConfiguration()["regexManagers"][0]
  );

  expect(res).toMatchSnapshot();
});

it("should find pre-commit dependencies", () => {
  const fileName = ".pre-commit-config.yaml";

  const res = extractPackageFile(
    loadFixture(fileName),
    fileName,
    loadRenovateConfiguration()["regexManagers"][1]
  );

  expect(res).toMatchSnapshot();
});

it("should find dependencies in pyproject.toml", () => {
  const fileName = "pyproject.toml";

  const res = extractPackageFile(
    loadFixture(fileName),
    fileName,
    loadRenovateConfiguration()["regexManagers"][1]
  );

  expect(res).toMatchSnapshot();
});
