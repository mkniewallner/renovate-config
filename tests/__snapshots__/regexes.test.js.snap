// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should find YAML dependencies 1`] = `
Object {
  "deps": Array [
    Object {
      "currentValue": "3.10.0",
      "datasource": "docker",
      "depName": "python",
      "replaceString": ": \\"3.10.0\\" # renovate: datasource=docker depName=python versioning=docker
",
      "versioning": "docker",
    },
    Object {
      "currentValue": "3.10.0",
      "datasource": "docker",
      "depName": "python",
      "replaceString": ": \\"3.10.0\\"  #renovate: datasource=docker depName=python versioning=docker
",
      "versioning": "docker",
    },
    Object {
      "currentValue": "1.1.11",
      "datasource": "pypi",
      "depName": "poetry",
      "replaceString": ": \\"1.1.11\\"   #   renovate: datasource=pypi depName=poetry
",
      "versioning": "semver",
    },
  ],
  "matchStrings": Array [
    "\\\\s*: \\"?(?<currentValue>.+?)\\"?[[:blank:]]+#[[:blank:]]*renovate: datasource=(?<datasource>[a-z-]+?) depName=(?<depName>.+?)(?: packageName=(?<packageName>.+?))?(?: versioning=(?<versioning>[a-z\\\\d-]+?))?\\\\s",
  ],
  "versioningTemplate": "{{#if versioning}}{{{versioning}}}{{else}}semver{{/if}}",
}
`;

exports[`should find dependencies in pyproject.toml 1`] = `
Object {
  "datasourceTemplate": "pypi",
  "deps": Array [
    Object {
      "currentValue": "==21.3.0",
      "datasource": "pypi",
      "depName": "argon2-cffi",
      "replaceString": "\\"argon2-cffi==21.3.0\\",  # renovate: pep440-python-dependency
",
    },
    Object {
      "currentValue": " == 1.2.1",
      "datasource": "pypi",
      "depName": "arrow",
      "replaceString": "\\"arrow == 1.2.1\\", #renovate: pep440-python-dependency
",
    },
    Object {
      "currentValue": " >= 0.16.0",
      "datasource": "pypi",
      "depName": "uvloop",
      "replaceString": "\\"uvloop >= 0.16.0\\", #renovate: pep440-python-dependency
",
    },
  ],
  "matchStrings": Array [
    "\\"(?<depName>[\\\\w-]+)(?<currentValue>.+?)\\",?[[:blank:]]+#[[:blank:]]*renovate: pep440-python-dependency\\\\s",
  ],
}
`;

exports[`should find pre-commit dependencies 1`] = `
Object {
  "datasourceTemplate": "pypi",
  "deps": Array [
    Object {
      "currentValue": "==21.9.1",
      "datasource": "pypi",
      "depName": "flake8-bugbear",
      "replaceString": "\\"flake8-bugbear==21.9.1\\"  # renovate: pep440-python-dependency
",
    },
    Object {
      "currentValue": "==0.14.1",
      "datasource": "pypi",
      "depName": "flake8-simplify",
      "replaceString": "\\"flake8-simplify==0.14.1\\" #renovate: pep440-python-dependency
",
    },
  ],
  "matchStrings": Array [
    "\\"(?<depName>[\\\\w-]+)(?<currentValue>.+?)\\",?[[:blank:]]+#[[:blank:]]*renovate: pep440-python-dependency\\\\s",
  ],
}
`;
